WatchFaceKotlin/wear/src/main/java/com/example/android/wearable/watchfacekotlin/watchface/AnalogComplicationWatchFaceService.kt

```diff
@@ -0,0 +1,834 @@
+
+        // Colors for all hands (hour, minute, seconds, ticks) based on photo loaded.
+        private var watchHandAndComplicationsColor = 0
+        private var watchHandHighlightColor = 0
+        private var watchHandShadowColor = 0
+        private var backgroundColor = 0
+
+
+        private lateinit var hourPaint: Paint
+        private lateinit var minutePaint: Paint
+        private lateinit var secondAndHighlightPaint: Paint
+        private lateinit var tickAndCirclePaint: Paint
+        private lateinit var backgroundPaint: Paint
+
+        /* Maps complication ids to corresponding ComplicationDrawable that renders the
+         * the complication data on the watch face.
+         */
+        private lateinit var complicationDrawableSparseArray: SparseArray<ComplicationDrawable>
+
+        // If active, this is used to set the background color to black.
+        private var backgroundComplicationActive = false
+
+
+        private var ambient = false
+        private var lowBitAmbient = false
+        private var burnInProtection = false
+
+        // Used to pull user's preferences for background color, highlight color, and visual
+        // indicating there are unread notifications.
+        private lateinit var sharedPreferences: SharedPreferences
+
+        // User's preference for if they want visual shown to indicate unread notifications.
+        private var unreadNotificationsPreference = false
+        private var numberOfUnreadNotifications = 0
+        private val timeZoneReceiver: BroadcastReceiver = object : BroadcastReceiver() {
+            override fun onReceive(context: Context, intent: Intent) {
+                calendar.timeZone = TimeZone.getDefault()
+                invalidate()
+            }
+        }
+
+        /**
+         * Coroutine/flow Job that animates the second hand every second when the watch face
+         * is active.
+         */
+        private var secondHandAnimationJob: Job? = null
+
+        override fun onCreate(holder: SurfaceHolder) {
+            Log.d(TAG, "onCreate")
+            super.onCreate(holder)
+
+            // Used throughout watch face to pull user's preferences.
+            val context = applicationContext
+            sharedPreferences = context.getSharedPreferences(
+                getString(R.string.analog_complication_preference_file_key),
+                MODE_PRIVATE
+            )
+            calendar = Calendar.getInstance()
+            setWatchFaceStyle(
+                WatchFaceStyle.Builder(this@AnalogComplicationWatchFaceService)
+                    .setAcceptsTapEvents(true)
+                    .setHideNotificationIndicator(true)
+                    .build()
+            )
+            loadSavedPreferences()
+            initializeComplicationsAndBackground()
+            initializeWatchFace()
+        }
+
+        // Pulls all user's preferences for watch face appearance.
+        private fun loadSavedPreferences() {
```

Comment:
Reading preferences feels outside of this responsibility of this object, can it be broken out?